### PR TITLE
PP-7417: Validation for Worldpay 3DS Flex Jwt Mac Key parameter

### PR DESCRIPTION
**Pull request by oswaldquek**

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class Worldpay3dsFlexJwtMacKeyValidatorTest {
+    
+    private Worldpay3dsFlexJwtMacKeyValidator validator = new Worldpay3dsFlexJwtMacKeyValidator();
+    
+    @Test
+    void test_for_valid_value() {
+        IntStream.range(1, 20).forEach(i -> assertTrue(validator.isValid(UUID.randomUUID().toString(), null)));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: Really not sure about having non-deterministic tests.

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks fine apart from the non-deterministic tests.

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java (hunk 1 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: I wanted to convince myself lower cased UUIDs were always produced but happy to remove this.

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: Some people like non-deterministic tests to find edge cases which I'm not against.

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class Worldpay3dsFlexJwtMacKeyValidatorTest {
+    
+    private Worldpay3dsFlexJwtMacKeyValidator validator = new Worldpay3dsFlexJwtMacKeyValidator();
+    
+    @Test
+    void test_for_valid_value() {
+        assertTrue(validator.isValid(UUID.randomUUID().toString(), null));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java:18]: I’m still sceptical of using randomness in the tests but I since I don’t think it’s likely we’ll have some valid values that pass and some valid values that fail, I’m OK with this.

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    
+    @Test
+    void test_for_valid_uppercase_UUID() {
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: Can we name this to indicate that upper-case values are considered invalid?

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        assertFalse(validator.isValid(UUID.randomUUID().toString().toUpperCase(), null));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "53f0917f101a442", "incorrect", "abcdef", "12345678901234567890123_" })
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: Can we have a couple of “near miss” invalid values here e.g. a UUID that’s 8-4-4-6-10 rather than 8-4-4-4-12, or a UUID that has the letter “h” in it?

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Just a couple of small test changes and I think this is good.

File: src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java (hunk 4 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccount/validation/Worldpay3dsFlexJwtMacKeyValidatorTest.java]: Good shout, done.

**Review by alexbishop1 — APPROVED** [review]
